### PR TITLE
[doc] Maximum execution time

### LINKED ISSUE #3125: Introduce Job-agnostic API to declare the maximal execution time for a Job
**What would you like to be added**:

A job-agnostic API to set the maximal execution time for a Job. 

There are some open questions:
- name of the API (I would initially propose annotation ` kueue.x-k8s.io/max-exec-time-seconds`)
- semantics of the deadline (measured since last admission or cumulative execution time) - I would initially suggest to measure the cumulative execution time across admissions , and re-use it in the EXEC_TIME command for kueuectl, but it will require API changes
- what happens after exceeding the time  - I would initially propose deactivating the workload, but alternatives might be worth considering

**Why is this needed**:

Different Jobs CRDs have a field with similar semantics, but there is no standard. For example batch/Job has `spec.activeDeadlneSeconds`, while JobSet does not have such an API for now.

We would like to have such an API to use it in kueuectl command as an analog of slurm's "-t" option. Long term (but out-of-scope here) we could use this value to optimize Job scheduling.

**Completion requirements**:

This enhancement requires the following artifacts:

- [ ] Design doc
- [ ] API change
- [ ] Docs update

The artifacts should be linked in subsequent comments.

### PR DESCRIPTION
**Pull request by trasc**
#### What type of PR is this?

/kind documentation

#### What this PR does / why we need it:

Add documentation for Maximum execution time.

#### Which issue(s) this PR fixes:

Fixes #3125 

#### Does this PR introduce a user-facing change?

```release-note
Added documentation for Maximum execution time.
```

### CODE REVIEW

**Comment by trasc** [comment]
/hold 
for  #3191 and #3200

**Comment by tenzen-y** [comment]
I will review this in the next week.

File: site/content/en/docs/reference/labels-and-annotations.md [hunk 1]
```diff
@@ -180,3 +180,14 @@ Example: `kueue.x-k8s.io/role-hash: "b54683bb"`
 Used on: [Plain Pods](/docs/tasks/run/plain_pods/).
 
 The annotation key is used as the name for a Workload podSet.
+
+
+### kueue.x-k8s.io/max-exec-time-seconds
```
Comment (mbobrovskyi) [inline site/content/en/docs/reference/labels-and-annotations.md:185]: We sorted this list in alphabetical order. Could you please move it after `kueue.x-k8s.io/managed`?

Comment (trasc) [reply site/content/en/docs/reference/labels-and-annotations.md]: done

**Review by mimowo — COMMENTED** [review]
/lgtm
/approve
Thanks!

**Comment by mimowo** [comment]
/release-note-note
I believe no need for release note as this is a new feature so one entry about it is enough,

**Comment by mimowo** [comment]
/release-note-edit
```release-note
NONE
```

**Pull request merged**